lib/rspec/core/configuration.rb

```diff
@@ -2033,6 +2032,12 @@ def on_existing_matching_groups(meta)
         end
       end
 
+      def add_hook_to_existing_groups(prepend_or_append, position, *meta, &block)
```

Comment:
Instead of adding a hook-specific method (which in turn delegates to some new logic added to the `Hooks` module for this case), is there a way to make `on_existing_matching_groups` handle nested groups properly?  It would be nice if the generic logic in `configuration` for applying metadata-based config to existing groups worked with nested groups so we don't have to solve issues like this in an ad-hoc manner for each metadata-based config thing.